data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/http/HttpConsumerVerticleFactory.java

```diff
@@ -110,25 +110,37 @@ public AbstractVerticle get(final DataPlaneContract.Resource resource,
       circuitBreakerOptions,
       egressConfig
     );
-    final var egressDeadLetterSender =
+
+    final Function<Vertx, ConsumerRecordSender<String, CloudEvent, HttpResponse<Buffer>>> egressDeadLetterSender =
       egressConfig == null || egressConfig.getDeadLetter() == null || egressConfig.getDeadLetter().isEmpty()
-        ? NO_DLQ_SENDER
+        ? vertx -> NO_DLQ_SENDER
         : createSender(egressConfig.getDeadLetter(), circuitBreakerOptions, egressConfig);
 
-    final var consumerOffsetManager = consumerRecordOffsetStrategyFactory
-      .get(consumer, resource, egress);
+    final Function<KafkaConsumer<String, CloudEvent>, ConsumerRecordOffsetStrategy<String, CloudEvent>>
+      consumerOffsetManager = c -> consumerRecordOffsetStrategyFactory.get(c, resource, egress);
```

Comment:
Are you wrapping a factory in a factory?! can you streamline a bit this code and remove those lambdas and place that code directly in consumerRecordHandlerFactory?

Reply:
I removed a bunch of unnecessary factories.